src/widgets/helper/SearchPopup.cpp

```diff
@@ -113,4 +129,78 @@ void SearchPopup::initLayout()
     }
 }
 
+std::vector<MessagePredicatePtr> SearchPopup::parsePredicates(
+    const QString &input)
+{
+    std::vector<MessagePredicatePtr> predicates;

```

Comment:
This should be a `std::vector<std::unique_ptr<MessagePredicate>>`  since the predicates aren't shared.

Reply:
Resolved in 5cd1683a.